astropy/wcs/wcs.py

```diff
@@ -636,7 +637,8 @@ def calc_footprint(self, header=None, undistort=True, axes=None, center=True):
                 naxis2 = header.get('NAXIS2', None)
 
         if naxis1 is None or naxis2 is None:
-            return None
+            raise ValueError(
+                    "image size could not be determined.")
```

Comment:
I agree raising an exception is better than returning None, not sure which one though. This looks fine to me.